src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php

```diff
@@ -124,6 +126,22 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
         self::assertThat(self::getRequest(), $constraint, $message);
     }
 
+    public static function assertThatForResponse(Response $response, Constraint $constraint, string $message = ''): void
+    {
+        try {
+            self::assertThat($response, $constraint, $message);
+        } catch (ExpectationFailedException $exception) {
+            if ($serverExceptionMessage = $response->headers->get('X-Debug-Exception')) {
+                $exception = new ExpectationFailedException(
+                    $exception->getMessage().PHP_EOL.'Server Exception: '.$serverExceptionMessage,
```

Comment:
I fear that changing the message might break existing test suites that expect a specific message.
Can you try this? How does it look like when a test fails?
```php
$exception->__construct($exception->getMessage(), $exception->getComparisonFailure(), new \Exception($serverExceptionMessage));
```

Reply:
That produces:

`Time: 374 ms, Memory: 20.00 MB

There was 1 failure:

1) App\Tests\Controller\HomeControllerTest::testC
Failed asserting that the Response has header "Content-Type" with value "application/json".

/srv/symfony/src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php:132
/srv/symfony/src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php:66
/srv/blog/tests/Controller/HomeControllerTest.php:29

Caused by
Exception: This a test exception. in /srv/symfony/src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php:139
Stack trace:
#0 /srv/symfony/src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php(66): Symfony\Bundle\FrameworkBundle\Test\WebTestCase::assertThatForResponse(Object(Symfony\Component\HttpFoundation\Test\Constraint\ResponseHeaderSame), '')
#1 /srv/blog/tests/Controller/HomeControllerTest.php(29): Symfony\Bundle\FrameworkBundle\Test\WebTestCase::assertResponseHeaderSame('Content-Type', 'application/jso...')
#2 /srv/blog/bin/.phpunit/phpunit-7.5-0/src/Framework/TestCase.php(1155): App\Tests\Controller\HomeControllerTest->testC()
etc...

Reply:
Great, that looks nice!